### PR TITLE
add methods GetCelebrityInfo and RecognizeCelebrities for Rekognition service

### PR DESCRIPTION
**Pull request by guillaumedelre**

### CODE REVIEW

**Comment by guillaumedelre** [comment]
Hello I think I need help, please let me know what is wrong or what I can improve.  Thank you very much.

**Comment by Nyholm** [comment]
Hey. I think most things look good. You need to add some tests. (see the generated templates)

FYI, the test for the SymfonyBundle is currently failing, it will be fixed in #992

**Comment by guillaumedelre** [comment]
@Nyholm thank you for your reply, i have added the test for RecognizeCelebritiesResponse, I have already written the test for GetCeebrityInfo. Tell me if somthing is still missing. Thank you

File: src/Service/Rekognition/tests/Unit/Input/GetCelebrityInfoRequestTest.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+namespace AsyncAws\Rekognition\Tests\Unit\Input;
+
+use AsyncAws\Core\Test\TestCase;
+use AsyncAws\Rekognition\Input\GetCelebrityInfoRequest;
+
+class GetCelebrityInfoRequestTest extends TestCase
+{
+    public function testRequest(): void
+    {
+        self::fail('Not implemented');
```
Comment (Nyholm) [inline src/Service/Rekognition/tests/Unit/Input/GetCelebrityInfoRequestTest.php:12]: This line needs to be removed

File: src/Service/Rekognition/tests/Unit/Input/RecognizeCelebritiesRequestTest.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+use AsyncAws\Core\Test\TestCase;
+use AsyncAws\Rekognition\Input\RecognizeCelebritiesRequest;
+use AsyncAws\Rekognition\ValueObject\Image;
+use AsyncAws\Rekognition\ValueObject\S3Object;
+
+class RecognizeCelebritiesRequestTest extends TestCase
+{
+    public function testRequest(): void
+    {
+        self::fail('Not implemented');
```
Comment (Nyholm) [inline src/Service/Rekognition/tests/Unit/Input/RecognizeCelebritiesRequestTest.php:14]: This needs to be implemented

**Review by Nyholm — COMMENTED** [review]
If you rebase the PR on master you will see if there is any issues. 

(Master is green now)

File: src/Service/Rekognition/tests/Unit/Input/GetCelebrityInfoRequestTest.php (hunk 1 above)
Comment (guillaumedelre) [reply src/Service/Rekognition/tests/Unit/Input/GetCelebrityInfoRequestTest.php]: done :)

File: src/Service/Rekognition/tests/Unit/Input/RecognizeCelebritiesRequestTest.php (hunk 2 above)
Comment (guillaumedelre) [reply src/Service/Rekognition/tests/Unit/Input/RecognizeCelebritiesRequestTest.php]: done :)

**Comment by guillaumedelre** [comment]
@Nyholm a check is still failing, how can I fix it ? (Checks / Generated output not changed (pull_request)

**Comment by Nyholm** [comment]
It is in the error message =)

`./generate --all`

**Comment by guillaumedelre** [comment]
@Nyholm thank you for your help !!!!!!

File: src/Service/Rekognition/tests/Unit/Input/GetCelebrityInfoRequestTest.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+use AsyncAws\Core\Test\TestCase;
+use AsyncAws\Rekognition\Input\GetCelebrityInfoRequest;
+
+class GetCelebrityInfoRequestTest extends TestCase
+{
+    public function testRequest(): void
+    {
+        $input = new GetCelebrityInfoRequest([
+            'Id' => 'change me',
```
Comment (Nyholm) [inline src/Service/Rekognition/tests/Unit/Input/GetCelebrityInfoRequestTest.php:13]: Could you add a more "real" value here.

File: src/Service/Rekognition/tests/Unit/Input/RecognizeCelebritiesRequestTest.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+{
+    public function testRequest(): void
+    {
+        $input = new RecognizeCelebritiesRequest([
+            'Image' => new Image([
+                'Bytes' => 'change me',
+                'S3Object' => new S3Object([
+                    'Bucket' => 'change me',
+                    'Name' => 'change me',
+                    'Version' => 'change me',
```
Comment (Nyholm) [inline src/Service/Rekognition/tests/Unit/Input/RecognizeCelebritiesRequestTest.php:20]: Same here, let's use values that look more like real values.

File: src/Service/Rekognition/tests/Unit/RekognitionClientTest.php [hunk 5]
```diff
@@ -94,6 +98,19 @@ public function testDetectFaces(): void
         self::assertFalse($result->info()['resolved']);
     }
 
+    public function testGetCelebrityInfo(): void
+    {
+        $client = new RekognitionClient([], new NullProvider(), new MockHttpClient());
+
+        $input = new GetCelebrityInfoRequest([
+            'Id' => 'change me',
```
Comment (Nyholm) [inline src/Service/Rekognition/tests/Unit/RekognitionClientTest.php:106]: Same

File: src/Service/Rekognition/tests/Unit/RekognitionClientTest.php [hunk 6]
```diff
@@ -124,6 +141,21 @@ public function testListCollections(): void
         self::assertFalse($result->info()['resolved']);
     }
 
+    public function testRecognizeCelebrities(): void
+    {
+        $client = new RekognitionClient([], new NullProvider(), new MockHttpClient());
+
+        $input = new RecognizeCelebritiesRequest([
+            'Image' => new Image([
+
```
Comment (Nyholm) [inline src/Service/Rekognition/tests/Unit/RekognitionClientTest.php:150]: Can we add at least one value here?

File: src/Service/Rekognition/tests/Unit/Result/GetCelebrityInfoResponseTest.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+use Psr\Log\NullLogger;
+use Symfony\Component\HttpClient\MockHttpClient;
+
+class GetCelebrityInfoResponseTest extends TestCase
+{
+    public function testGetCelebrityInfoResponse(): void
+    {
+        // see https://docs.aws.amazon.com/rekognition/latest/APIReference/API_GetCelebrityInfo.html
+        $response = new SimpleMockedResponse('{
+            "Name": "Tim Berners-Lee",
```
Comment (Nyholm) [inline src/Service/Rekognition/tests/Unit/Result/GetCelebrityInfoResponseTest.php:18]: Great!

**Review by Nyholm — APPROVED** [review]
Thank you. Just some minors

File: src/Service/Rekognition/tests/Unit/Input/GetCelebrityInfoRequestTest.php (hunk 3 above)
Comment (guillaumedelre) [reply src/Service/Rekognition/tests/Unit/Input/GetCelebrityInfoRequestTest.php:13]: done

File: src/Service/Rekognition/tests/Unit/Input/RecognizeCelebritiesRequestTest.php (hunk 4 above)
Comment (guillaumedelre) [reply src/Service/Rekognition/tests/Unit/Input/RecognizeCelebritiesRequestTest.php]: done

File: src/Service/Rekognition/tests/Unit/RekognitionClientTest.php (hunk 5 above)
Comment (guillaumedelre) [reply src/Service/Rekognition/tests/Unit/RekognitionClientTest.php]: done

File: src/Service/Rekognition/tests/Unit/RekognitionClientTest.php (hunk 6 above)
Comment (guillaumedelre) [reply src/Service/Rekognition/tests/Unit/RekognitionClientTest.php]: done

File: src/Service/Rekognition/tests/Unit/RekognitionClientTest.php [hunk 8]
```diff
@@ -147,7 +147,7 @@ public function testRecognizeCelebrities(): void
...
...GvL9tv1KY0dn4UUc3+NP2VGNWyyoX551IXdamnpUGy3KOMwrUZ2zElzeOB2yHqXYFwelPpKjjNm3ctHLuksZQvgtM4mPDuJUZqP1LUhiZ61yfpVfHYVxd3d6OrkMTENavpWFFDdnZWIjJ25fiWlIN2b9WpXuy+HjASIxrdqEh9lVZBzejOzdK6IKkZuhSozNRV5g4OrW6ZmUP6WWgMBn9NE0mXDN1GOXDqTRnlRvspkuxhN34ourlki5uHdTcPSkUhb/z9tMR5otpkmGJ3avmQMjuUhejh0qO1k0eXFAkgvp+nupcmUgiz5qMaehMA48FYhLKvdFVx9NKMmEqNkyQcFoZaVU13nVW+no0qSGWn6ECLG9o7N+zyqaEm89FVE3zZk0ZPSuabQNl4S/3rU2D2glwpMcRE1C5elYcZf7lNf/yqURKmqP1B7mPu3Cdom9C6oyX6C2PtsMQLELtVfzjwuKcHYxe0h1CQr7Z7lvuzkBDHO9H73eTas4p4nHfsfsFCwuzfagMSLUdrluqDIEIQgAQhCQAhCFQHhL1CEgFQmSfpTA9QhCABVpolZXhCgiUVLkzSw/xJfBVoWo3Suzj9l8DDxWH+JZ5RehdJiIFlyQfErizzuowbmeMTqxDErAwKaOL0q2znxYdybCgrgglhCimtWJ7mLHsKIKQEIUHUlQI86EIKBInQgBEy9NIgATJUIAEIQgAQheCgD1CEJ0B5cvUISAEJUIJsZCVCBjJUJECsdIi9LcgQ16L1Hei5AxiJeLy5eIA9uS3pb0GqECW5KSUkAMRqOQkGoySsTVkMnnWfiIq+ZaBKGSJWpHHkw6jJKKqN0tAsOo90q9ocq6aiiWHdHg9FoWfEo5BU+0NP+KqMeaCqqlhfStiSJVygVLIcsul34MmTBM/mRHhacFpbhMOHZHtSY9Cr4M8cLRG6qtCQGUPBL2h1rpUUyif0I3StFRRyUUvIaLpkinJEoZIlcJVyU+0B9P8AAqjEpoxTpL0azWGBImHJSXKrvVGUqjWdahRaKVQ71QlKoSlS1FKBNJKoSnUchKEjU6ylDzO8EkXqEioi+qg6SxGrEaqxkrAIAmy86/JP9rTta54gMIxVHCBvZY+7NP8A0441+qtoYpgByJ7RAbyIukA51/Pnt52jfG4rEYknu8IlI7pO4GgP4a5sr4R29Mqbl9P1/wAbHLzFwr
```
[the remark this answers is not in the document]
Comment (Nyholm) [reply src/Service/Rekognition/tests/Unit/RekognitionClientTest.php:150]: Dont miss this comment =)

[the remark this answers is not in the document]
Comment (guillaumedelre) [reply src/Service/Rekognition/tests/Unit/RekognitionClientTest.php]: done

**Review by Nyholm — APPROVED** [review]

**Comment by jderusse** [comment]
Thank you @guillaumedelre

**Pull request merged**